Grbl_Esp32/src/Motors/StandardStepper.h

```diff
@@ -5,19 +5,24 @@
 namespace Motors {
     class StandardStepper : public Motor {
     public:
-        StandardStepper();
         StandardStepper(uint8_t axis_index, uint8_t step_pin, uint8_t dir_pin, uint8_t disable_pin);
 
         virtual void config_message();
         virtual void init();
-        virtual void set_direction_pins(uint8_t onMask);
+        virtual void set_direction(bool);
```

Comment:
Hmm the 'virtual's seem wrong here. I would expect 'override' at some methods tho.

Reply:
I need to read and reread and reread again all the C++ rules for method inheritance.  There are so many different options...

Reply:
C++ inheritance and overloading is very confusing t.b.h. That's why most people only use a small, KISS subset, which avoids bugs. Google and many others even enforce these in their coding style guide:

1. Classes should always use public inheritance. E.g. `class Foo : public Bar`. 
2. Don't add 'virtual' in the base class for methods that you don't want overridden. Prefer protected virtual to public virtual methods. 
3. Add `virtual void Foo() = 0;` for things that must always be overridden (that's called "pure virtual"). If not, it will give a compiler error. 
4. Don't add `virtual` in derived classes for overriding; use `override`. E.g. `void Foo() override {... }`. If done incorrectly, compiler error.
5. If you have a virtual, you should have a `virtual ~Destructor() {}`. This enforces proper cleanup of derived classes by 'delete'. If omitted, you might have a leak.

So, public inheritance, virtual base class, if the method is '{}', it should probably be pure virtual, otherwise virtual, use override in derived classes, add destructor in all classes. That's it.

Reply:
Thanks, that advice is very helpful